db/migrations/000006_create_environment_reservations_table.up.sql

```diff
@@ -0,0 +1,14 @@
+CREATE TABLE IF NOT EXISTS environment_reservations(
+    id                serial PRIMARY KEY,
+    environment_id    integer REFERENCES environments (id),
+    name              text,
+    source            text,
+    url               text,
+    ttl               interval, 
+    mutex             boolean,
+    metadata          jsonb,
```

Comment:
I can remove this, this was an artifact from when this was still job_requests and the table had larger scope.

Reply:
actaully I think there are two parts of data here. since there's intent to reuse environments ala allocation pools, there's two things pieces of data we have here, Environment.customization would show the current state of the environement, metadata would show the desired state of the environment ala all fiabs getting nuked and repopulated with expected app versions every run.

Reply:
>  Environment.customization would show the current state of the environement, metadata would show the desired state of the environment ala all fiabs getting nuked and repopulated with expected app versions every run

Having trouble wrapping my head around the distinction here but this is a really great point.

It sounds like we want to capture:
* **actual deployed versions**. For this it would probably make sense to re-use the Builds/Deploys endpoints Mike is implementing for lead time tracking, although we didn't initially plan on using those to track PR builds and deploys 
* **desired versions**. In other words, "preview env 1234 should deploy Leo off `my-feature-branch` in the Leo repo." The trick here is that _how_  those desired versions get wired in to Helm/ArgoCD seems to be really up in the air. Wondering if makes to omit this from this PR until there's more clarity. Especially since ArgoCD is developing this useful tool https://argocd-image-updater.readthedocs.io/en/stable/ which could change how we plug things in.

We *for sure* know that environment soft deletes and allocation pools are needed, and those seem like big/important enough features for one PR haha